src/test/java/nextstep/subway/unit/FarePoliciesTest.java

```diff
@@ -15,58 +22,43 @@ class FarePoliciesTest {
     @BeforeEach
     void setup() {
         //given 요금 정책을 확정한다.
-        FarePolicy base = new FixedFarePolicy(1250);
-        FarePolicy ten = new BetweenUnitFarePolicy(10, 50, 5, 100);
-        FarePolicy fifth = new GreaterUnitFarePolicy(50, 8, 100);
         farePolicies = new FarePolicies();
-        farePolicies.addPolicies(base, ten, fifth);
     }
 
-    @Test
-    @DisplayName("8km")
-    void km_8() {
-        //then 8km면 요금은 1250원이다.
-        assertThat(farePolicies.calculate(8)).isEqualTo(1250);
-    }
+    @ParameterizedTest
+    @DisplayName("거리 비례 요금 계산")
+    @CsvSource(value = {"8,1250", "10,1250", "12,1350", "16,1450", "50,2050", "60,2250", "65, 2250"})
+    void calculateFare(int distance, int fare) {
+        //then 거리를 넣으면 요금이 계산된다.
 
-    @Test
-    @DisplayName("10km")
-    void km_10() {
-        //then 10km면 요금은 1250원이다.
-        assertThat(farePolicies.calculate(10)).isEqualTo(1250);
-    }
-    @Test
-    @DisplayName("12km")
-    void km_12() {
-        //then 12km면 요금은 1350원이다.
-        assertThat(farePolicies.calculate(12)).isEqualTo(1350);
-    }
-    @Test
-    @DisplayName("16km")
-    void km_16() {
-        //then 8km면 요금은 1450원이다.
-        assertThat(farePolicies.calculate(16)).isEqualTo(1450);
-    }
-    @Test
-    @DisplayName("50km")
-    void km_50() {
-        //then 8km면 요금은 2050원이다.
-        assertThat(farePolicies.calculate(50)).isEqualTo(2050);
+        CalculateConditions conditions = new CalculateConditionBuilder(distance).build();
+
+        assertThat(farePolicies.calculate(conditions)).isEqualTo(fare);
     }
 
-    @Test
-    @DisplayName("60km")
-    void km_60() {
-        //then 8km면 요금은 2250원이다.
-        assertThat(farePolicies.calculate(60)).isEqualTo(2250);
+    @ParameterizedTest
+    @CsvSource(value = {"6, 850", "12, 850", "13, 1360", "18, 1360", "19, 2050", "20, 2050"})
+    @DisplayName("연령별 요금 할인 계산")
+    void ageDiscount(int age, int fare) {
+        //when 연령을 계산 조건으로 설정한다.
+        CalculateConditions conditions = new CalculateConditionBuilder(50)
+                .age(age)
+                .build();
+        //then 요금이 할인된다.
+        assertThat(farePolicies.calculate(conditions)).isEqualTo(fare);
```

Comment:
```suggestion
        //given 연령을 계산 조건으로 설정한다.
        CalculateConditions conditions = new CalculateConditionBuilder(50)
                .age(age)
                .build();

        //then 요금을 계산한다.
        int result = farePolicies.calculate(conditions)
        
        //then 할인된 요금을 리턴한다.
        assertThat(result).isEqualTo(fare);
```
BDD에서 사용하는 각 절은 아래와 같은 의미를 가지는데요,

- Given: 시나리오 상에서 주어진 환경을 정의합니다.
- When: 사용자가 어떤 행위를 하는 것을 정의합니다.
- Then: 그에 따른 어떠한 결과를 정의합니다. (검증)

테스트하려는 것을 when절에 적어주시면 나머지는 명확해지더라구요.
conditions는 calculate()를 위해 주어져야하는 값으로 given절이 되고 farePolicies.calculate()를 호출하는게 when 절이 되겠네요!